test/extended/cli/version.go

```diff
@@ -0,0 +1,61 @@
+package cli
+
+import (
+        "fmt"
+        "log"
+        "os"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/openshift-tests/test/extended/util"
+)
+
+var _ = g.Describe("[cli]oc version[Conformance]", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc                     = exutil.NewCLI("oc-rsh", exutil.KubeConfigPath())
+	)
+
+	g.Describe("oc version before login", func() {
+		g.It("check version without login", func() {
+			g.By("check version info with default KUBECONFIG") 
+			out, err := oc.Run("version").Args("-o", "yaml").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(out).To(o.ContainSubstring("serverVersion"))
+
+			g.By("check version info without --kubeconfig,~/.kube/config,KUBECONFIG") 
+                        kubeconfig :=exutil.KubeConfigPath()
+                        defer os.Setenv("KUBECONFIG", kubeconfig)
+                        os.Setenv("KUBECONFIG", "")
+			out, err = oc.Run("version").Args("-o", "yaml").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(out).To(o.ContainSubstring("clientVersion"))
+                        
+			g.By("check version info with empty kubeconfig")
+                        emptyFile, err := os.Create("/tmp/emptykubeconfig")
+	                if err != nil {
+		                log.Fatal(err)
+	                } 
+                        defer closeFile(emptyFile)
+			out, err = oc.Run("version").Args("--kubeconfig", "/tmp/emptykubeconfig").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(out).To(o.ContainSubstring("Client Version"))
```

Comment:
Can we check version string with a regular expression or whatever kind of pattern matching? We have seen in the past many changes like adding.removing `v` in front of the version sting and other minor changes that broke version parsing.

btw for YAML format we also better check path and version sting format with pattern. Will save us a lot of potential troubles down the road.

Reply:
@akostadinov thanks for review , will try to complete all the request.

Reply:
@akostadinov  Have updated , could you please review again ? thanks .